simulator.py

```diff
@@ -187,9 +189,9 @@ def reset(self, config: CartPoleConfig = CartPoleConfig()) -> tuple[State, float
         Returns:
             Returns (initial_state, initial_target).
         '''
-        log.info(f'reset')
+        LOGGER.info(f'reset')
 
-        self.config = config
+        self.config = config or Config()
```

Comment:
Config() as default value is mutable, therefore all calls to reset without config argument will have the same mutable object as default. this can be very bad if config gets altered later